### PR TITLE
functionMap: additions, improvement, corrections

### PR DESCRIPTION
**Pull request by IanDelMar**
Most of the PR deals with the `$output` argument and conditional return types handling `$output ∈ {OBJECT, ARRAY_A, ARRAY_N}`.

Additions: `wpdb::get_row()`, `wpdb::get_results()`, `get_bookmark()`, `get_category()` and `get_category_by_path()`. I added tests, but did not test each parameter variation or unions.

### CODE REVIEW

File: functionMap.php [hunk 1]
```diff
@@ -13,38 +13,37 @@
...
+    'get_objects_in_term' => [null, 'args' => 'array{order?: string}'],
+    'have_posts' => [null, '@phpstan-impure' => ''],
+    'rawurlencode_deep' => ['T', '@phpstan-template' => 'T', 'value' => 'T'],
+    'sanitize_category' => ['T', '@phpstan-template' => 'T of array|object', 'category' => 'T'],
+    'sanitize_post' => ['T', '@phpstan-template' => 'T of array|object', 'post' => 'T'],
+    'sanitize_term' => ['T', '@phpstan-template' => 'T of array|object', 'term' => 'T'],
+    'stripslashes_deep' => ['T', '@phpstan-template' => 'T', 'value' => 'T'],
+    'urldecode_deep' => ['T', '@phpstan-template' => 'T', 'value' => 'T'],
+    'urlencode_deep' => ['T', '@phpstan-template' => 'T', 'value' => 'T'],
+    'wp_clear_scheduled_hook' => ['(0|positive-int|($wp_error is false ? false : \WP_Error))', 'args'=>$cronArgsType],
```
Comment (johnbillion) [inline functionMap.php:35]: I wasn't aware PHPStan supported this syntax for conditional return types, nice

File: wordpress-stubs.php [hunk 2]
```diff
@@ -130182,6 +130191,7 @@ function get_tax_sql($tax_query, $primary_table, $primary_id_column)
      * @param string             $filter   Optional. How to sanitize term fields. Default 'raw'.
      * @return WP_Term|array|WP_Error|null WP_Term instance (or array) on success, depending on the `$output` value.
      *                                     WP_Error if `$taxonomy` does not exist. Null for miscellaneous failure.
+     * @phpstan-param 'OBJECT'|'ARRAY_A'|'ARRAY_N' $output
      * @phpstan-return ($output is 'ARRAY_A' ? array<string, string|int>|\WP_Error|null : ($output is 'ARRAY_N' ? list<string|int>|\WP_Error|null : \WP_Term|\WP_Error|null))
      */
     function get_term($term, $taxonomy = '', $output = \OBJECT, $filter = 'raw')
```
Comment (szepeviktor) [inline wordpress-stubs.php:130197]: core says: constant
phpstan says: string
viktor says: WordPress
```php
define( 'OBJECT', 'OBJECT' );
```

**Review by szepeviktor — APPROVED** [review]

File: functionMap.php (hunk 1 above)
Comment (IanDelMar) [reply functionMap.php:35]: Me neither. It's great! It helps to simplify some of the conditions and make them easier to read.

File: wordpress-stubs.php (hunk 2 above)
Comment (IanDelMar) [reply wordpress-stubs.php:130197]: To be fair, core also says string but passes the default string as a constant. But yeah, `define( 'OBJECT', 'OBJECT' );` is weird.

**Comment by szepeviktor** [comment]
A blind merge comes here ...

**Pull request merged**